### PR TITLE
fix: restore camelCase support in setPrefixedStyle

### LINKED ISSUE #1046: [BUG] setPrefixedStyle silently fails for camelCase CSS property
### Describe the bug
`setPrefixedStyle` in [styleUtils.ts](https://github.com/maxGraph/maxGraph/blob/main/packages/core/src/util/styleUtils.ts#L120) uses `style.setProperty(name, value)`, but that function [requires kebab-case/hyphen case property names](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/setProperty#propertyname) (e.g. `transform-origin`). The only camelCase call site is `setPrefixedStyle(this.textarea.style, 'transformOrigin', '0px 0px')` in [CellEditorHandler.ts](https://github.com/maxGraph/maxGraph/blob/main/packages/core/src/view/plugin/CellEditorHandler.ts#L649), which is a silent no-op. This was introduced in commit 61648e43c, which changed `setPrefixedStyle` from `style[name] = value` (accepts camelCase) to `style.setProperty(name, value)` (requires kebab-case).

### To Reproduce
1. Create a graph with at least one vertex
2. Zoom to a non-1x level
3. Double-click a cell to start editing
4. The editor overlay is visibly offset from the cell label

### Expected behavior
`transform-origin: 0px 0px` should be applied to the cell editor textarea. The editor overlay should align with the cell label at all zoom levels. This likely worked correctly in `mxGraph`.

### Screenshots
 N/A

### Environment
- `maxGraph` version or commit: present on current main
- Desktop or mobile: Desktop
- OS and version: Linux
- Browser and version: Latest Chromium

### Additional context
Simplest fix is probably to change the call site to kebab-case/hyphen case: `setPrefixedStyle(this.textarea.style, 'transform-origin', '0px 0px');`.

### PR DESCRIPTION
**Pull request by redfish4ktc**
## Problem

`setPrefixedStyle` writes CSS properties with `CSSStyleDeclaration.setProperty`, which [only accepts kebab-case names](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/setProperty#propertyname). Any camelCase name is lowercased, matches no known property, and is silently discarded.

The reported symptom is the cell editor: `CellEditorHandler` calls `setPrefixedStyle(this.textarea.style, 'transformOrigin', '0px 0px')`, so the textarea never receives its transform origin and the editing overlay is visibly offset from the cell label at any zoom level other than 1x.

The vendor prefixed write was broken too, for **every** input spelling. The prefix is built by capitalizing the first character of the name, which produces webkit-cased names such as `WebkitTransformOrigin`. Those are CSSOM IDL attribute spellings, not CSS property names, so `setProperty` discards them as well. Feeding the function a kebab-case name does not help either: it yields the meaningless `WebkitTransform-origin`. Consequently the prefixed write in `RubberBandHandler` (`'transition'`) has also been dead, even though its standard write happens to work, single word names surviving the lowercasing intact.

## Root cause

Commit 61648e43c, a large "Converting *Handlers into plugins. Keep resolving errors" refactor, changed exactly two lines of this function:

```diff
-  style[name] = value;
+  style.setProperty(name, value);
   if (prefix !== null && name.length > 0) {
     name = prefix + name.substring(0, 1).toUpperCase() + name.substring(1);
-    style[name] = value;
+    style.setProperty(name, value);
```

`style[name] = value` does not compile here (`TS7015: Element implicitly has an 'any' type because index expression is not of type 'number'`), so this was a type error being silenced during a mass refactor, not a deliberate API change. The kebab-case requirement was never chosen, and the JSDoc has kept documenting camelCase ever since.

## Fix

Restore the property assignment used by `mxGraph`, with the cast that commit was avoiding:

```typescript
const properties = style as unknown as Record<string, string>;
properties[name] = value;

if (!isNullish(prefix) && name.length > 0) {
  name = prefix + name.substring(0, 1).toUpperCase() + name.substring(1);
  properties[name] = value;
}
```

Property assignment reaches the CSSOM camel-cased, dashed **and** webkit-cased attributes, so both writes work again. A comment records why the assignment is deliberate, so it does not get "fixed" back to `setProperty`.

Both call sites are left untouched: their existing camelCase arguments become correct again, rather than needing to change.

CSS custom properties are the one case that must keep using `setProperty`, so they get an early return before the prefix logic:

```typescript
if (name.startsWith('--')) {
  style.setProperty(name, value);
  return;
}
```

Unlike standard properties, a custom property has no attribute on `CSSStyleDeclaration`, so assigning one would create a JavaScript property and no CSS declaration at all, and it is never vendor prefixed. This capability is not inherited from `mxGraph`, which used assignment and had the same limitation: it appeared as a side effect of 61648e43c, the very commit that introduced the bug fixed here. It has been available for four years, so it is preserved rather than silently removed.

Vendor prefixing is deliberately kept rather than removed. It is very likely obsolete (`transform-origin` and `transition` have been unprefixed for over a decade, the browser detection relies on user agent sniffing, and the project targets modern browsers only), but the impact on external consumers calling this exported utility for other properties is unknown, so removing it belongs in its own change.

## Documented limitation

The mangling only produces a valid name from camelCase input, which the JSDoc now states explicitly:

> **WARNING**: `name` must be written in camelCase, as in the example above.
>
> A kebab-case name such as `transform-origin` still sets the standard property, but silently skips the vendor prefixed one.
>
> A CSS custom property such as `--overlay-offset` is set as is, and is never vendor prefixed.

## Tests

The function had no test coverage. 14 cases added in `packages/core/__tests__/util/styleUtils.test.ts`, each written before the corresponding fix and confirmed to fail against the previous implementation (7 failures for the camelCase standard write plus every vendor prefixed write, then 3 more for the custom properties):

- standard property set from camelCase, kebab-case and single word names
- no vendor property written when no prefix applies
- both properties written on Safari, Chrome and Firefox
- single word names get prefixed too, covering the dead `RubberBandHandler` write
- Webkit prefix wins over Moz when both flags are set
- the `name.length > 0` guard for an empty name
- the kebab-case limitation, asserting the malformed `WebkitTransform-origin`, so the JSDoc warning has an executable counterpart
- custom properties set with and without a vendor prefix active, and never given a prefixed variant, asserting `cssText` is exactly `--overlay-offset: 10px;` so a JavaScript property masquerading as a CSS declaration cannot pass

Two implementation notes. `Client.IS_SF` is `true` by default under jsdom, so each test sets the browser flags explicitly instead of inheriting the environment; those flags are global mutable state, captured before any test runs and restored after each one. And jsdom implements no vendor prefixed CSS property, so a prefixed assignment lands as a plain JavaScript property on the declaration, which a helper reads. That is the only way this path is observable in jsdom, and it works precisely because the fix uses assignment rather than `setProperty`.

## Validation

- `npm test -w packages/core`: 535 tests passed, 60 suites
- `npm run test-check -w packages/core` and `tsc --noEmit`: clean
- `npm run lint`: clean

:heavy_check_mark: Not covered by automated tests: the visual outcome. The tests prove `transform-origin` reaches the textarea style, not that the overlay aligns on screen. Manual check for a reviewer: open Storybook, zoom to a non-1x level, double-click a cell, and confirm the editor lines up with the label.

Fixes #1046

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved application of standard and vendor-prefixed styles across supported browsers.
  * Added reliable support for camelCase properties and CSS custom properties.
  * Improved prefix selection and handling of empty or missing property names.
  * Ensured style updates use the appropriate browser-compatible format.

* **Tests**
  * Expanded coverage for browser-specific styling, prefix precedence, property naming formats, custom properties, and edge cases.
  * Added checks to ensure browser-state behavior remains consistent.

### CODE REVIEW

File: packages/core/src/util/styleUtils.ts [hunk 1]
```diff
@@ -103,25 +104,36 @@ export const parseCssNumber = (value: string) => {
...
+  // and therefore silently discards both the camelCase names documented above and the vendor prefixed names
+  // computed below. See https://github.com/maxGraph/maxGraph/issues/1046
+  const properties = style as unknown as Record<string, string>;
+  properties[name] = value;
 
-  if (prefix !== null && name.length > 0) {
+  if (!isNullish(prefix) && name.length > 0) {
     name = prefix + name.substring(0, 1).toUpperCase() + name.substring(1);
-    style.setProperty(name, value);
+    properties[name] = value;
```
[the remark this answers is not in the document]
Comment (redfish4ktc) [reply packages/core/src/util/styleUtils.ts:144]: Confirmed and fixed in 58989969e.

Verified in jsdom before changing anything, rather than assuming:

| write | `getPropertyValue('--overlay-offset')` | `cssText` |
|---|---|---|
| `setProperty('--overlay-offset', '10px')` | `10px` | `--overlay-offset: 10px;` |
| `style['--overlay-offset'] = '11px'` | `""` | `""` (JavaScript property only) |

Custom properties have no attribute on `CSSStyleDeclaration`, unlike standard properties, which is exactly why `transform-origin` works through assignment (CSSOM dashed attribute) while `--overlay-offset` does not.

Implemented as suggested: `--` names are routed through `setProperty` with an early return, before the prefix logic, since custom properties are never vendor prefixed.

```typescript
if (name.startsWith('--')) {
  style.setProperty(name, value);
  return;
}
```

Three regression tests added, written first and confirmed failing against the previous state: the property is set both with and without a vendor prefix active, and no prefixed variant is produced. They read the value back with `getPropertyValue('--overlay-offset')`, and also assert that `cssText` is exactly `--overlay-offset: 10px;` so that a JavaScript property masquerading as a CSS declaration cannot pass.

One nuance worth recording: this capability is not inherited from `mxGraph`, which used property assignment and had the same limitation. It appeared as a side effect of the switch to `setProperty` in 61648e43c, the very commit that introduced the bug this PR fixes. It has been available for four years though, so it is preserved rather than silently removed.
